include/fiction/algorithms/simulation/sidb/does_cds_match_logic_for_input_pattern.hpp

```diff
@@ -0,0 +1,68 @@
+//
+// Created by Jan Drewniok on 12.11.24.
+//
+
+#ifndef FICTION_DOES_CDS_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
+#define FICTION_DOES_CDS_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
+
+#include "fiction/algorithms/simulation/sidb/is_operational.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+
+#include <kitty/traits.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <utility>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * Checks if a given charge distribution correctly encodes the expected logic for a specified input pattern,
+ * based on a provided truth table.
+ *
+ * @note Kinks are rejected.
+ *
+ * Example:
+ * In the ground state charge distribution of an AND gate, kinks are rejected for the gate to be considered operational.
+ * Given an input pattern of `1`, this function will:
+ * - Verify that the left input wire encodes `0`.
+ * - Verify that the right input wire encodes `1`.
+ * - Verify that the output wire encodes `0`.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @tparam TT Truth table type.
+ * @param cds Charge distribution surface, containing charge state information for each SiDB.
+ * @param params The parameters used to determine if a layout is `operational` or `non-operational`.
+ * @param spec Expected Boolean function of the layout given as a multi-output truth table.
+ * @param input_pattern The specific input pattern of the given charge distribution surface.
+ * @param input_and_output_bdl_wire Pair of input and output BDL wires.
+ * @return The operational status indicating if the charge distribution matches the logic for the given input pattern.
+ */
+template <typename Lyt, typename TT>
+[[nodiscard]] operational_status does_cds_match_logic_for_input_pattern(
```

Comment:
I'm not sure this name change went in the right direction. Worst case, people don't know what a CDS is and get confused. I was thinking along the lines of `validate_logic_encoding` or `verify_logic_match`. A function name should describe **what** a function is doing. The parameters it is doing it on don't need to be part of the name because those will be clear given the parameter list. 

Example: you would call a function `add` if it adds two numbers together. Probably not `add_two_numbers_of_which_one_or_both_can_be_negative_and_return_an_int`. I'm exaggerating of course to make the point.